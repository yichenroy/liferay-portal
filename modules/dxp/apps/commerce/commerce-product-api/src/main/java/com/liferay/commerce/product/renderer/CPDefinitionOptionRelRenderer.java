/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}