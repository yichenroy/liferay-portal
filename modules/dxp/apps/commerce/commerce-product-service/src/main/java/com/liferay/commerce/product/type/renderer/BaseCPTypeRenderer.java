/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}