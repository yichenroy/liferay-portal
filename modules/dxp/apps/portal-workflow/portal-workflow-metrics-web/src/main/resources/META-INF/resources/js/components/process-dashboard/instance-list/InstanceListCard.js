import React, { Fragment } from 'react';
import { AppContext } from '../../AppContext';
import autobind from 'autobind-decorator';
import InstanceListFilter from './InstanceListFilter';
import InstanceListTable from './InstanceListTable';
import ListView from '../../../shared/components/list/ListView';
import PaginationBar from '../../../shared/components/pagination/PaginationBar';
import { parse } from '../../../shared/components/router/queryString';
import ReloadButton from '../../../shared/components/list/ReloadButton';

class InstanceListCard extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			error: null,
			items: [],
			loading: false,
			totalCount: 0
		};
	}

	componentWillMount() {
		this.loadProcess();
	}

	componentWillReceiveProps(nextProps) {
		this.loadInstances(nextProps);
	}

	getRequestUrl({ page, pageSize, processId, query } = this.props) {
		const requestUrl = `/processes/${processId}/instances?page=${page}&pageSize=${pageSize}`;

		const filters = parse(query).filters;

		if (typeof filters === 'object') {
			return (
				requestUrl +
				Object.keys(filters).reduce(
					(acc, cur) => `${acc}&${cur}=${filters[cur]}`,
					''
				)
			);
		}

		return requestUrl;
	}

	@autobind
	handleRequestError() {
		this.setState({
			error: Liferay.Language.get(
				'there-was-a-problem-retrieving-data-please-try-reloading-the-page'
			),
			loading: false
		});
	}

	loadInstances(props = this.props) {
		const { loading } = this.state;

		if (loading) {
			return;
		}

		const requestUrl = this.getRequestUrl(props);

		return this.requestData(requestUrl)
			.then(({ items, totalCount }) => {
				this.setState({
					items,
					totalCount
				});
			})
			.catch(this.handleRequestError);
	}

	loadProcess({ processId } = this.props) {
		const { loading } = this.state;

		if (loading) {
			return;
		}

		return this.requestData(`/processes/${processId}/title`)
			.then(data => {
				this.context.setTitle(`${data}: ${Liferay.Language.get('all-items')}`);
			})
			.catch(this.handleRequestError);
	}

	requestData(endpoint) {
		const { client } = this.context;

		this.setState({
			loading: true
		});

		return client.get(endpoint).then(({ data }) => {
			this.setState({
				error: null,
				loading: false
			});

			return data;
		});
	}

	render() {
		const { error, items = [], loading, totalCount } = this.state;
		const { page, pageSize, processId, query } = this.props;

		const fetching = !loading && !totalCount;

		return (
			<Fragment>
				<InstanceListFilter
					processId={processId}
					query={query}
					totalCount={totalCount}
				/>

				<div className="container-fluid-1280 mt-4">
					<ListView
						emptyActionButton={<ReloadButton />}
						emptyMessageText={Liferay.Language.get(
							'there-are-no-process-items-at-the-moment'
						)}
						errorMessageText={error}
						fetching={fetching}
						loading={loading}
					>
						<InstanceListTable items={items} />

						<PaginationBar
							page={page}
							pageCount={items.length}
							pageSize={pageSize}
							totalCount={totalCount}
						/>
					</ListView>
				</div>
			</Fragment>
		);
	}
}

InstanceListCard.contextType = AppContext;
export default InstanceListCard;