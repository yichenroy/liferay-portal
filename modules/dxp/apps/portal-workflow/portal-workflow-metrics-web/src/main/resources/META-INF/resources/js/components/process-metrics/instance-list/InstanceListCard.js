import {
	InstanceListProvider,
	InstanceListContext
} from './store/InstanceListStore';
import React, {useContext} from 'react';
import {ErrorContext} from '../../../shared/components/request/Error';
import {getFiltersParam} from '../../../shared/components/filter/util/filterUtil';
import {InstanceFiltersProvider} from './store/InstanceFiltersStore';
import InstanceItemDetail from './InstanceItemDetail';
import InstanceListFilters from './InstanceListFilters';
import InstanceListTable from './InstanceListTable';
import ListView from '../../../shared/components/list/ListView';
import {LoadingContext} from '../../../shared/components/request/Loading';
import PaginationBar from '../../../shared/components/pagination/PaginationBar';
import ReloadButton from '../../../shared/components/list/ReloadButton';
import Request from '../../../shared/components/request/Request';

export function InstanceListCard({page, pageSize, processId, query}) {
	const filters = getFiltersParam(query);
	const {
		slaStatuses = [],
		statuses = [],
		taskKeys = [],
		timeRange = []
	} = filters;

	return (
		<Request>
			<InstanceFiltersProvider
				processId={processId}
				processStatusKeys={statuses}
				processStepKeys={taskKeys}
				slaStatusKeys={slaStatuses}
				timeRangeKeys={timeRange}
			>
				<InstanceListProvider
					page={page}
					pageSize={pageSize}
					processId={processId}
					query={query}
				>
					<InstanceListCard.Header
						processId={processId}
						query={query}
					/>

					<InstanceListCard.Body
						page={page}
						pageSize={pageSize}
						processId={processId}
					/>
				</InstanceListProvider>
			</InstanceFiltersProvider>
		</Request>
	);
}

InstanceListCard.Body = ({page, pageSize, processId}) => {
	const {error} = useContext(ErrorContext);
	const {items = [], totalCount} = useContext(InstanceListContext);
	const {loading} = useContext(LoadingContext);

	const fetching = !loading && !totalCount;
	const errorMessage = error
		? Liferay.Language.get(
				'there-was-a-problem-retrieving-data-please-try-reloading-the-page'
		  )
		: null;

	return (
		<>
			<div className="container-fluid-1280 mt-4">
				<ListView
					emptyActionButton={<ReloadButton />}
					emptyMessageText={Liferay.Language.get(
						'there-are-no-process-items-at-the-moment'
					)}
					errorMessageText={errorMessage}
					fetching={fetching}
					loading={loading}
				>
					<InstanceListTable items={items} />

					<PaginationBar
						page={page}
						pageCount={items.length}
						pageSize={pageSize}
						totalCount={totalCount}
					/>
				</ListView>
			</div>

			<InstanceItemDetail processId={processId} />
		</>
	);
};

InstanceListCard.Header = () => {
	const {totalCount} = useContext(InstanceListContext);

	return (
		<Request.Success>
			<InstanceListFilters totalCount={totalCount} />
		</Request.Success>
	);
};

export default InstanceListCard;
