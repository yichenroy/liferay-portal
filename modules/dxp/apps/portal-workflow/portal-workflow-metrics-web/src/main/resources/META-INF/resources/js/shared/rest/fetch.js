/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import axios from 'axios';

/**
 * @module fetch
 * @description Basic alias of fetch client to perform consults in portal rest API.
 * @example
 * import fetch from '@/shared/rest/fetch';
 * fetch.get('/process').then(res => console.log(res));
 */
const restClient = axios.create({
	baseURL: '/o/portal-workflow-metrics/v1.0'
});

axios.defaults.headers.common[
	'Accept-Language'
] = Liferay.ThemeDisplay.getBCP47LanguageId();

axios.defaults.params = {
	['p_auth']: Liferay.authToken
};

export default restClient;
