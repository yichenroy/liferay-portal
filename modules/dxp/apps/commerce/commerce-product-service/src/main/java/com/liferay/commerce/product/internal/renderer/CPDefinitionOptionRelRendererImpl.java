/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}