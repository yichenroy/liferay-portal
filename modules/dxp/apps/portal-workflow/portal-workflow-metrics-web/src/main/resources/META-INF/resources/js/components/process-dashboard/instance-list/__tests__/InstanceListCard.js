import fetch from '../../../../test/mock/fetch';
import fetchFailure from '../../../../test/mock/fetchFailure';
import InstanceListCard from '../InstanceListCard';
import React from 'react';
import renderer from 'react-test-renderer';
import { MockRouter as Router } from '../../../../test/mock/MockRouter';

test('Should build request url with query filters', () => {
	const data = {
		items: [
			{
				assetTitle: 'Item Subject Test',
				assetType: 'Process',
				dateCreated: new Date('2019', '01', '01'),
				id: 12351,
				status: 'Overdue',
				taskNames: ['Step 1', 'Step 2', 'Step 3'],
				userName: 'User Test'
			}
		],
		title: 'Single Approver',
		totalCount: 1
	};

	const component = mount(
		<Router client={fetch(data)}>
			<InstanceListCard
				page={1}
				pageSize={10}
				processId={12351}
				query="?filters.slaStatus%5B0%5D=overdue"
			/>
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	const requestUrl = instance.getRequestUrl();

	expect(requestUrl).toEqual(
		'/processes/12351/instances?page=1&pageSize=10&slaStatus=overdue'
	);
});

test('Should component load process instances', () => {
	const data = {
		items: [
			{
				assetTitle: 'Item Subject Test',
				assetType: 'Process',
				dateCreated: new Date('2019', '01', '01'),
				id: 12351,
				status: 'Overdue',
				taskNames: ['Step 1', 'Step 2', 'Step 3'],
				userName: 'User Test'
			}
		],
		title: 'Single Approver',
		totalCount: 1
	};

	const component = mount(
		<Router client={fetch(data)}>
			<InstanceListCard page={1} pageSize={10} processId={12351} />
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.state.loading = false;

	instance.loadInstances().then(data => {
		expect(data).toEqual(data);
	});
});

test('Should component receive props', () => {
	const data = {
		items: [
			{
				assetTitle: 'Item Subject Test',
				assetType: 'Process',
				dateCreated: new Date('2019', '01', '01'),
				id: 12351,
				status: 'Overdue',
				taskNames: ['Step 1', 'Step 2', 'Step 3'],
				userName: 'User Test'
			}
		],
		title: 'Single Approver',
		totalCount: 1
	};

	const component = mount(
		<Router client={fetch(data)}>
			<InstanceListCard
				page={1}
				pageSize={10}
				processId={12351}
				query="?filters.slaStatus%5B0%5D=overdue"
			/>
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.componentWillReceiveProps({
		...instance.props,
		page: 2,
		query: 'filters.slaStatus%5B0%5D=overdue&filters.slaStatus%5B1%5D=ontime'
	});

	expect(component).toMatchSnapshot();
});

test('Should component set error state after request fails', () => {
	const component = mount(
		<Router client={fetchFailure()}>
			<InstanceListCard page={1} pageSize={10} processId={35315} />
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.state.loading = false;

	instance.loadInstances().catch(() => {
		expect(instance.state.errors).toEqual(
			'There was a problem retrieving data. Please try reloading the page.'
		);
	});
});

test('Should component shows empty state when items is undefined', () => {
	const component = mount(
		<Router client={fetch({})}>
			<InstanceListCard />
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.setState({
		items: undefined
	});

	expect(component).toMatchSnapshot();
});

test('Should not reload component while loading state is true', () => {
	const component = mount(
		<Router client={fetch()}>
			<InstanceListCard page={1} pageSize={10} processId={35315} />
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.state.loading = true;

	const result = instance.loadInstances({ ...instance.props, page: 2 });

	expect(result).toBeNil();
});

test('Should not reload process while loading state is true', () => {
	const component = mount(
		<Router client={fetch()}>
			<InstanceListCard page={1} pageSize={10} processId={35315} />
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.state.loading = true;

	const result = instance.loadProcess({ ...instance.props });

	expect(result).toBeNil();
});

test('Should render component', () => {
	const data = {
		items: [
			{
				assetTitle: 'Item Subject Test',
				assetType: 'Process',
				dateCreated: new Date('2019', '01', '01'),
				id: 12351,
				status: 'Untracked',
				taskNames: ['Step 1', 'Step 2', 'Step 3'],
				userName: 'User Test'
			}
		],
		title: 'Single Approver',
		totalCount: 1
	};

	const component = mount(
		<Router client={fetch(data)}>
			<InstanceListCard page={1} pageSize={10} processId={12351} />
		</Router>
	);

	const instance = component.find(InstanceListCard).instance();

	instance.componentWillReceiveProps({ page: 2 });

	expect(component).toMatchSnapshot();
});

test('Should render component with empty data', () => {
	const data = {
		items: null,
		totalCount: 0
	};

	const component = renderer.create(
		<Router client={fetch(data)}>
			<InstanceListCard processId={35315} />
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component with error state', () => {
	const component = renderer.create(
		<Router client={fetchFailure()}>
			<InstanceListCard processId={35315} />
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});