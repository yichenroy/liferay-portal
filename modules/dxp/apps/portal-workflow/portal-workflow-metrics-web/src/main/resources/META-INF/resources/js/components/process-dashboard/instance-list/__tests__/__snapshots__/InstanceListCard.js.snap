// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component receive props 1`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <InstanceListCard
        page={1}
        pageSize={10}
        processId={12351}
      >
        <div
          className="container-fluid-1280 mt-4"
        >
          <ListView
            emptyActionButton={<ReloadButton />}
            emptyMessageText="There are no process items at the moment."
            errorMessageText={null}
            fetching={false}
            loading={true}
          >
            <div>
              <div
                className="border-1 pb-6 pt-6 sheet"
              >
                <LoadingState>
                  <span
                    aria-hidden="true"
                    className="loading-animation"
                  />
                </LoadingState>
              </div>
            </div>
          </ListView>
        </div>
      </InstanceListCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should component shows empty state when items is undefined 1`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <InstanceListCard>
        <div
          className="container-fluid-1280 mt-4"
        >
          <ListView
            emptyActionButton={<ReloadButton />}
            emptyMessageText="There are no process items at the moment."
            errorMessageText={null}
            fetching={false}
            loading={true}
          >
            <div>
              <div
                className="border-1 pb-6 pt-6 sheet"
              >
                <LoadingState>
                  <span
                    aria-hidden="true"
                    className="loading-animation"
                  />
                </LoadingState>
              </div>
            </div>
          </ListView>
        </div>
      </InstanceListCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should render component 1`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <InstanceListCard
        page={1}
        pageSize={10}
        processId={12351}
      >
        <div
          className="container-fluid-1280 mt-4"
        >
          <ListView
            emptyActionButton={<ReloadButton />}
            emptyMessageText="There are no process items at the moment."
            errorMessageText={null}
            fetching={false}
            loading={true}
          >
            <div>
              <div
                className="border-1 pb-6 pt-6 sheet"
              >
                <LoadingState>
                  <span
                    aria-hidden="true"
                    className="loading-animation"
                  />
                </LoadingState>
              </div>
            </div>
          </ListView>
        </div>
      </InstanceListCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should render component with empty data 1`] = `
<div
  className="container-fluid-1280 mt-4"
>
  <div>
    <div
      className="border-1 pb-6 pt-6 sheet"
    >
      <span
        aria-hidden="true"
        className="loading-animation"
      />
    </div>
  </div>
</div>
`;

exports[`Should render component with error state 1`] = `
<div
  className="container-fluid-1280 mt-4"
>
  <div>
    <div
      className="border-1 pb-6 pt-6 sheet"
    >
      <span
        aria-hidden="true"
        className="loading-animation"
      />
    </div>
  </div>
</div>
`;
